practice.py

```diff
@@ -1,17 +1,42 @@
-from TexSoup import TexSoup
 from jinja2 import Environment
 from jinja2 import PackageLoader
 from jinja2 import Template
 
+def generator(lines):
+	"""
+	>>> lines = ['Question: 1', 'Option: 1', 'Question: 2', 'Option: 3']
+	>>> questions = list(generator(lines))
+	>>> len(questions)
+	2
+	>>> len(questions[0])
+	2
+	>>> len(questions[1])
+	2
+	"""
+	lst, seen = [], 0
+	for line in lines:
+		if line.startswith('Question:') and seen == 0:
+			lst.append(line)
+			seen += 1
+		elif line.startswith('Question:') and seen == 1:
+			val = lst
+			lst = []
+			lst.append(line)
+			yield val
+		elif line.strip():
+			lst.append(line)
+	yield lst
```

Comment:
this could be simplified:

```
lst = []
for line in lines:
     if line.starstwith('Question:'):
         if lst:
             yield lst
         lst.append(line)
     elif line.strip():
         lst.append(line)
if lst:
     yield lst
```